### ISSUE TITLE
Cumulative gradient？Using small BatchSize to simulate big BatchSize 

### ISSUE THREAD

**Issue by liugl7** [open-mmlab/mmcv#190]
Could you develop the "Cumulative gradient"？
It using small BatchSize to simulate big BatchSize.

eg: [PyTorch中在反向传播前为什么要手动将梯度清零? ](https://www.zhihu.com/question/303070254/answer/573037166)

**Comment by hellock** [comment]
You can implement a new optimizer hook by clearing the gradients every several iterations. Ref: https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/hooks/optimizer.py#L16.

**Comment by liugl7** [comment]
OK，thank you!

##########################################################
# Copyright (c) Open-MMLab. All rights reserved.
from torch.nn.utils import clip_grad

from .hook import Hook


class OptimizerHook(Hook):

     def __init__(self, grad_clip=None, every_n_iters=1):
        self.grad_clip = grad_clip
        self.every_N_iters = every_n_iters

    def clip_grads(self, params):
        clip_grad.clip_grad_norm_(
            filter(lambda p: p.requires_grad, params), **self.grad_clip)

    def after_train_iter(self, runner):
        runner.outputs['loss'].backward()
        if (runner.iter + 1) % self.every_N_iters == 0:
            if self.grad_clip is not None:
                self.clip_grads(runner.model.parameters())
            runner.optimizer.step()
            runner.optimizer.zero_grad()
